devito/ir/iet/visitors.py

```diff
@@ -174,19 +181,23 @@ class CGen(Visitor):
     Return a representation of the Iteration/Expression tree as a :module:`cgen` tree.
     """
 
-    def __init__(self, *args, compiler=None, **kwargs):
+    def __init__(self, *args, printer=None, **kwargs):
         super().__init__(*args, **kwargs)
-        self._compiler = compiler
-
-    # The following mappers may be customized by subclasses (that is,
-    # backend-specific CGen-erators)
-    _qualifiers_mapper = {
-        'is_const': 'const',
-        'is_volatile': 'volatile',
-        '_mem_constant': 'static',
-        '_mem_shared': '',
-    }
-    _restrict_keyword = 'restrict'
+        if printer is None:
+            from devito.passes.iet.languages.C import CPrinter
```

Comment:
when do we end up here?

Reply:
In a plain `Node`, i.e here:
https://github.com/devitocodes/devito/blob/661939b34500e7ee608f6b3b479045c76ef1e7c8/devito/ir/iet/nodes.py#L105